### PR TITLE
update spillover

### PR DESCRIPTION
**Pull request by mikejiang**
Here is what I did

```
$ diff -crB flowStats_4.5.3/flowStats/ flowStats_4.5.4/flowStats/ >patch.diff
$ patch -p2 < ~/Downloads/patch.diff
```

### CODE REVIEW

File: tests/testthat/test-spillover.R [hunk 1]
```diff
@@ -29,13 +30,19 @@ test_that("spillover: Match columns using ordered", {
   ref_file <- system.file("extdata", "compdata", "compref2", package="flowCore")
   comp_ref <- as.matrix(read.table(ref_file, check.names = FALSE))
```
Comment (amcdavid) [inline tests/testthat/test-spillover.R]: These look like they are defined below with the `structure`
```suggestion
```

File: tests/testthat/test-spillover.R [hunk 2]
```diff
@@ -98,7 +105,9 @@ test_that("spillover_ng: Using preconstructed flowSet with filenames as sample n
 
 test_that("spillover_ng: Using preconstructed flowSet with channels in new order", {
   match <- read.csv(matchfile_path, colClasses = "character")
+  ref_file <- system.file("extdata", "compdata", "compref4", package="flowCore")
```
Comment (amcdavid) [inline tests/testthat/test-spillover.R]: Defined on lines 132/133
```suggestion
```

File: tests/testthat/test-spillover.R [hunk 3]
```diff
@@ -98,7 +105,9 @@ test_that("spillover_ng: Using preconstructed flowSet with filenames as sample n
...
   fs <- read.flowSet(files=match$filename, path=control_path)
+  comp_ref <- as.matrix(read.table(ref_file, check.names = FALSE))
```
Comment (amcdavid) [inline tests/testthat/test-spillover.R]:
```suggestion
```

**Review by gfinak — APPROVED** [review]

File: R/spillover.R [hunk 4]
```diff
@@ -191,32 +195,43 @@ setMethod("spillover",
...
+                solution=clue::solve_LSAP(t(dists))
+                channel_order <- as.vector(solution)
+                for(i in 1:ncol(dists)){
+                    cat("matching ",colnames(dists)[i]," to ", rownames(dists)[channel_order[i]],"\n")
                 }
               }else{
                 # Just bump-down the channel_order to account for
                 # the removal of the unstained row. 
                 channel_order <- channel_order - (channel_order > unstained)
+                for(i in 1:ncol(inten)){
```
Comment (amcdavid) [inline R/spillover.R]:
```suggestion
                for(i in seq_len(ncol(inten))){
```

File: R/spillover.R [hunk 5]
```diff
@@ -191,32 +195,43 @@ setMethod("spillover",
...
+                    cat("matching ",colnames(inten)[i]," to ", rownames(inten)[channel_order[i]],"\n")
```
Comment (amcdavid) [inline R/spillover.R]:
```suggestion
                    message("matching ",colnames(inten)[i]," to ", rownames(inten)[channel_order[i]])
```

File: R/spillover.R [hunk 6]
```diff
@@ -191,32 +195,43 @@ setMethod("spillover",
...
               # Updates the "rownames" of the intensity matrix. If the channel
               # order was not set above, then a guess is made based on the
               # largest statistic for the compensation control (i.e., the row).
               if (any(is.na(channel_order))) {
-                channel_order <- apply(inten, 1, which.max)
-                if (anyDuplicated(channel_order) > 0) {
-                  stop("Unable to match stains with controls based on intensity: ",
-                       "a single stain matches to several multiple controls. ",
-                       call. = FALSE)
+                dists = sapply(colnames(inten),function(x)adist(x,rownames(inten)))
```
Comment (amcdavid) [inline R/spillover.R:203]: Taking the edit distance between the `colnames(inten)` and `rownames(inten)`--could be reasonable, but if that's what were doing the above comment is obsolete.

**Review by amcdavid — APPROVED** [review]
A few nits / Q's we could try to resolve to help our future selfs.

File: R/spillover.R [hunk 7]
```diff
@@ -191,32 +195,43 @@ setMethod("spillover",
...
+                colnames(dists)=colnames(inten)
+                rownames(dists)=rownames(inten)                
+                solution=clue::solve_LSAP(t(dists))
```
Comment (amcdavid) [inline R/spillover.R]: Taking the edit distance between the `colnames(inten)` and `rownames(inten)`--could be reasonable, but if that's what were doing the above comment is obsolete.

Also: need to add `clue` to the DESCRIPTION.

**Pull request merged**